syntax = "proto3";

package posts;

service postMaker{
  rpc makePost(makePostRequest) returns (makePostResponse);
}

service postGetter{
  rpc getPost(GetPostRequest) returns(GetPostResponse);
  rpc GetPostPaginated(GetPostRequest) returns (GetPostPaginatedResponse);
  rpc GetFirstPostId (GetFirstPostIdRequest) returns (GetFirstPostIdResponse);
  rpc CheckPostId (CheckPostIdRequest) returns (CheckPostIdResponse);
}


message makePostRequest{
  string file_name = 1;
  bytes photo_bytes = 2;
  string post_title = 3;
  string post_description = 4;
  string seller_contact = 5;
  int64 user_id = 6;

}

message makePostResponse{
  int32 code = 1;
  string state = 2;
}

message GetPostRequest{
  int64 post_id = 1;
}

message GetPostResponse{
  int32 code = 1;
  string state = 2;

  bytes photo_bytes = 3;
  string post_title = 4;
  string post_description = 5;
  string seller_contact = 6;
  string creation_time = 7;
  int32 post_id = 8;
  int64 user_id = 9;
}

message GetPostPaginatedResponse{
  repeated GetPostResponse posts = 1 ;
}

message GetFirstPostIdRequest{

}
message GetFirstPostIdResponse{
  int32 first_post_id = 1;
  string state = 2;
  int32 code = 3;
}

message CheckPostIdRequest{
  int32 post_id = 1;
}
message CheckPostIdResponse{
  bool state = 1;
}